src/Adapter/State/CommandHandler/EditStateHandler.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\State\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\State\Command\EditStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\State\CommandHandler\EditStateHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\CannotUpdateStateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateNotFoundException;
+use PrestaShopException;
+use State;
+
+/**
+ * Handles state editing
+ */
+class EditStateHandler extends AbstractStateHandler implements EditStateHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CannotUpdateStateException
+     * @throws StateConstraintException
+     * @throws StateException
+     * @throws StateNotFoundException
+     */
+    public function handle(EditStateCommand $command): void
+    {
+        $this->updateStateFromCommandData(
+            $this->getState($command->getStateId()),
+            $command
+        );
+    }
+
+    /**
+     * @param State $state
+     * @param EditStateCommand $command
+     *
+     * @throws CannotUpdateStateException
+     * @throws StateConstraintException
+     * @throws StateException
+     */
+    private function updateStateFromCommandData(State $state, EditStateCommand $command)
```

Comment:
My guess is readability again. By splitting a single portion of code into smaller functions, even if they are used only once, it might become easier for a developer to read the code because the name of the private method will give some explanation about what's going on.

Reply:
Fixed